sdks/python/apache_beam/io/gcp/bigquery_tools.py

```diff
@@ -651,9 +651,19 @@ def wait_for_bq_job(self, job_reference, sleep_duration_sec=5, max_retries=0):
           job_reference.projectId, job_reference.jobId, job_reference.location)
       logging.info('Job %s status: %s', job.id, job.status.state)
       if job.status.state == 'DONE' and job.status.errorResult:
-        raise RuntimeError(
-            'BigQuery job {} failed. Error Result: {}'.format(
-                job_reference.jobId, job.status.errorResult))
+        if 'Retrying may solve the problem.' in job.status.errorResult.message:
```

Comment:
If I understood correctly, at this point the job is already completed as "DONE" and with error.

self.get_job(...) is just a GET call to get. the BigQuery load job status. It won't retry the load job.

The current retry logic is to poll existing job, and this poll API call can intermittently fail and get retried

Reply:
So retry poll job when a job status returned DONE with errorResult, it won't actually retry the load job.

Reply:
Yes, you are right. Close this.